vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -88,11 +88,11 @@ public void close() {
     requireNonNull(resultHandler, "resultHandler cannot be null");
 
     MongoCollection<JsonObject> coll = getCollection(collection, writeOption);
-    String id = document.getString(ID_FIELD);
+    Object id = document.getValue(ID_FIELD);
     if (id == null) {
       coll.insertOne(document, convertCallback(resultHandler, wr -> document.getString(ID_FIELD)));
```

Comment:
No, it can be leaved as is. If `_id` is not present, MongoDB takes case of it and creates a String-parseable ObjectId, so no signature change is needed.